### PR TITLE
feat(EMS-1849-1850): Your business -  Companies house page

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* Added companies house search page
* Removed companies house search part from companies details page
* Updated tests
* Improved e2e tests for company details
* Added unit and e2e tests for companies house search page

### CODE REVIEW

File: e2e-tests/pages/your-business/company-details/companyDetails.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  [WEBSITE]: {
+    label: () => cy.get(`[data-cy="${WEBSITE}-label"]`),
+    input: () => cy.get(`[data-cy="${WEBSITE}"]`),
+    errorMessage: () => cy.get(`[data-cy="${WEBSITE}-error"]`),
+  },
+  [PHONE_NUMBER]: {
+    label: () => cy.get(`[data-cy="${PHONE_NUMBER}-label"]`),
+    input: () => cy.get(`[data-cy="${PHONE_NUMBER}"]`),
+    hint: () => cy.get(`[data-cy="${PHONE_NUMBER}-hint"]`),
+    errorMessage: () => cy.get(`[data-cy="${PHONE_NUMBER}-error"]`),
```
Comment (ttbarnes) [inline e2e-tests/pages/your-business/company-details/companyDetails.js:44]: One selector for all of them? 


```js
field: (id) => ({
  label: ...
  input: ...
});
```

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js [hunk 2]
```diff
@@ -59,28 +58,8 @@ describe('Insurance - Your business - Company details page - Save and go back',
...
-  });
-
-  describe(`when required fields are completed and ${INPUT} is entered incorrectly`, () => {
-    it('should not display validation errors and redirect to task list', () => {
-      cy.navigateToUrl(url);
-
-      cy.completeCompanyDetailsForm({ companiesHouseNumber: '**/*' });
-
+      companyDetails[TRADING_NAME].yesRadioInput().click();
+      companyDetails[TRADING_ADDRESS].yesRadioInput().click();
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js:62]: Could we use `yesRadioInput` from `pages/shared` instead and delete this from `companyDetails`? I think we have a tech ticket to do this

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js [hunk 3]
```diff
@@ -164,11 +140,10 @@ describe('Insurance - Your business - Company details page - Save and go back',
     it('should render valid submitted values when going back to the page', () => {
       cy.navigateToUrl(url);
 
-      companyDetails.companiesHouseSearch().should('have.value', COMPANIES_HOUSE_NUMBER);
-      companyDetails.tradingNameYesRadioInput().should('be.checked');
-      companyDetails.tradingAddressYesRadioInput().should('be.checked');
-      companyDetails.phoneNumber().should('have.value', VALID_PHONE_NUMBERS.LANDLINE.NORMAL);
-      companyDetails.companyWebsite().should('have.value', WEBSITE_EXAMPLES.VALID);
+      companyDetails[TRADING_NAME].yesRadioInput().should('be.checked');
+      companyDetails[TRADING_ADDRESS].yesRadioInput().should('be.checked');
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js:144]: Same as above, for all this file, whilst you're in here?

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-submit.spec.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        expectedErrors,
+        COMPANY_DETAILS_ERRORS[PHONE_NUMBER].INCORRECT_FORMAT,
+      );
+    });
+  });
+
+  describe('continue to next page', () => {
+    const companyDetailsFormVariables = {
+      phoneNumber: VALID_PHONE_NUMBERS.LANDLINE.NORMAL,
+      website: WEBSITE_EXAMPLES.VALID,
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-submit.spec.js:92]: Use field IDs here?

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/validation/company-details-trading-name-validation.spec.js [hunk 5]
```diff
@@ -35,9 +41,7 @@ describe("Insurance - Your business - Company details page- As an Exporter I wan
 
     cy.navigateToUrl(url);
 
-    cy.keyboardInput(companyDetails.companiesHouseSearch(), COMPANIES_HOUSE_NUMBER);
-
-    companyDetails.tradingAddressYesRadioInput().click();
+    companyDetails[TRADING_ADDRESS].yesRadioInput().click();
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/validation/company-details-trading-name-validation.spec.js:44]: radio input from `pages/shared`?

File: src/ui/server/controllers/insurance/business/companies-house-number/index.ts [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+      submittedValues,
+      ...pageVariables(application.referenceNumber),
+    });
+  } catch (err) {
+    console.error('Error getting companies house number page %O', err);
+    return res.redirect(PROBLEM_WITH_SERVICE);
+  }
+};
+
+const redirectToExitPage = {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/companies-house-number/index.ts:91]: Seems this isn't used anymore?

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js (hunk 2 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js]: done

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js (hunk 3 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/company-details-save-and-back.spec.js]: done

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/validation/company-details-trading-name-validation.spec.js (hunk 5 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/your-business/company-details/validation/company-details-trading-name-validation.spec.js:44]: done

File: e2e-tests/commands/insurance/complete-company-details-form.js [hunk 7]
```diff
@@ -1,23 +1,31 @@
...
  * does not submit the form
  * @param {Object} companyDetailsVariables - companiesHouseNumber, phoneNumber and website
  */
-const completeCompaniesDetailsForm = ({ companiesHouseNumber, phoneNumber, website }) => {
-  cy.keyboardInput(companyDetails.companiesHouseSearch(), companiesHouseNumber);
-  companyDetails.tradingNameYesRadioInput().click();
-  companyDetails.tradingAddressYesRadioInput().click();
+const completeCompaniesDetailsForm = ({ phoneNumber, website }) => {
+  companyDetails[TRADING_NAME].yesRadioInput().click();
+  companyDetails[TRADING_ADDRESS].yesRadioInput().click();
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/complete-company-details-form.js:21]: radios from `pages/shared`?

File: e2e-tests/commands/insurance/assert-change-answers-page-url.js [hunk 8]
```diff
@@ -9,14 +9,16 @@ import { INSURANCE_ROOT } from '../../constants/routes/insurance';
...
+const assertChangeAnswersPageUrl = (referenceNumber, route, fieldId, fragmentSuffix, noSuffix = false) => {
   let expected = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${route}`;
 
   if (fieldId) {
     expected = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${route}#${fieldId}`;
 
     if (fragmentSuffix) {
       expected = `${expected}-${fragmentSuffix}`;
+    } else if (noSuffix) {
+      expected = `${expected}`;
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/assert-change-answers-page-url.js:21]: Why is this needed? Feels strange, when changing an answer it should have a suffix so it goes to the label/heading of the form input

Comment (Zainzzkk) [reply e2e-tests/commands/insurance/assert-change-answers-page-url.js:21]: the company details page has a heading.  It should focus on the heading which has data-cy of `heading` so it shouldn't be heading-label which is what happens otherwise

Comment (ttbarnes) [reply e2e-tests/commands/insurance/assert-change-answers-page-url.js:21]: Sounds like it can be achieved by passing `heading` to `fragmentSuffix` instead of adding a new param and condition here?

Comment (Zainzzkk) [reply e2e-tests/commands/insurance/assert-change-answers-page-url.js:21]: that will make it `heading-heading` and if I leave it out it will be `heading-label`

File: e2e-tests/commands/insurance/complete-company-details-form.js (hunk 7 above)
Comment (Zainzzkk) [reply e2e-tests/commands/insurance/complete-company-details-form.js:21]: done

**Review by ttbarnes — APPROVED** [review]
